crates/oxc_linter/src/rules/react/jsx_no_target_blank.rs

```diff
@@ -0,0 +1,658 @@
+) -> (bool, &'a str, bool, bool) {
+    let default = (false, "", false, false);
+    match expr {
+        Expression::StringLiteral(str) => (check_rel_val(str, allow_referrer), "", false, false),
+        Expression::ConditionalExpression(expr) => {
+            let consequent = match_rel_expression(&expr.consequent, allow_referrer);
+            let alternate = match_rel_expression(&expr.alternate, allow_referrer);
+            if let Expression::Identifier(identifier) = &expr.test {
+                return (
+                    consequent.0 && alternate.0,
+                    identifier.name.as_str(),
+                    consequent.0,
+                    alternate.0,
+                );
+            }
+            (consequent.0 && alternate.0, "", consequent.0, alternate.0)
+        }
+        _ => default,
+    }
+}
+
+fn check_rel<'a>(
+    attribute_value: &'a JSXAttributeValue<'a>,
+    allow_referrer: bool,
+) -> (bool, &'a str, bool, bool) {
+    let default = (false, "", false, false);
+    match attribute_value {
+        JSXAttributeValue::StringLiteral(str) => {
+            (check_rel_val(str, allow_referrer), "", false, false)
+        }
+        JSXAttributeValue::ExpressionContainer(expr) => match &expr.expression {
+            JSXExpression::Expression(expr) => match_rel_expression(expr, allow_referrer),
+            JSXExpression::EmptyExpression(_) => default,
+        },
+        _ => default,
+    }
+}
+
+fn match_target_expression<'a>(expr: &'a Expression<'a>) -> (bool, &'a str, bool, bool) {
+    let default = (false, "", false, false);
+    match expr {
+        Expression::StringLiteral(str) => {
+            (str.value.as_str().to_lowercase() == "_blank", "", false, false)
+        }
+        Expression::ConditionalExpression(expr) => {
+            let consequent = match_target_expression(&expr.consequent);
+            let alternate = match_target_expression(&expr.alternate);
+            if let Expression::Identifier(identifier) = &expr.test {
+                return (
+                    consequent.0 || alternate.0,
+                    identifier.name.as_str(),
+                    consequent.0,
+                    alternate.0,
+                );
+            }
+            (consequent.0 || alternate.0, "", consequent.0, alternate.0)
+        }
+        _ => default,
+    }
+}
+
+fn check_target<'a>(attribute_value: &'a JSXAttributeValue<'a>) -> (bool, &'a str, bool, bool) {
+    let default = (false, "", false, false);
+    match attribute_value {
+        JSXAttributeValue::StringLiteral(str) => {
+            (str.value.as_str().to_lowercase() == "_blank", "", false, false)
+        }
+        JSXAttributeValue::ExpressionContainer(expr) => {
+            if let JSXExpression::Expression(expr) = &expr.expression {
+                match_target_expression(expr)
+            } else {
+                default
+            }
+        }
+        _ => default,
+    }
+}
+
+impl Rule for JsxNoTargetBlank {
```

Comment:
What about moving this part below the `declare_oxc_lint`, we can see the main part of this rule at first glance.